### DISCUSSION TITLE
Synchronizing with Web3Auth Status

### DISCUSSION THREAD

**Discussion by robport** [Web3Auth/Web3Auth#649] [Help]
How can I synchronise with the connection and authentication status of the Web3Auth.  For example to get the balance of the default account immediately the authentication is completed?

In particular, I want to listen to the web3auth.status changing.  If the status is not_ready, I want to show an 'initialising...' message, and when it goes to either ready or connected, I would show the login and logged in views respectively.

ATM, I can see no way to listen to the event that is emitted when the web3auth moves from not_ready to ready.

**Comment by rinzler99** [comment]
Pls refer to this docs: https://web3auth.io/docs/sdk/web/core/initialize#adapter-events

**Comment by robport** [comment]
Thanks.   However, the ready event is never emitted. e.g.

const web3auth = new Web3Auth({
  clientId: clientId,
  chainConfig: {
    chainNamespace: CHAIN_NAMESPACES.SOLANA,
    rpcTarget: rpcTarget,
  }
});

web3auth.on(ADAPTER_EVENTS.READY, () => {
  console.error('ready');
});

I see that the state changes to Ready (via a setTimeout), but the
event is not emitted.

Can you point me to a demo repo where the Ready event is working?

Cheers,

Rob

**Reply by rinzler99** [reply]
`const web3auth = new Web3Auth({
  clientId: clientId,
  chainConfig: {
    chainNamespace: CHAIN_NAMESPACES.SOLANA,
    rpcTarget: rpcTarget,
  }
});

web3auth.on("ready", () => {
  console.error('ready');
});
`

Can you try passing "ready" instead of ADAPTER_EVENTS.READY?

**Reply by robport** [reply]
Done.  Didn't work.  I've added you to the repo.  Perhaps it will be obvious where I am going wrong.

https://github.com/project-excalibur/web3auth-demo

If you refresh the page when the user is logged out, you will see that the status never changes from not_ready.

**Reply by rinzler99** [reply]
went through your code, you can refer [this](https://github.com/Web3Auth/Web3Auth/blob/master/demo/auth0-react-app/app/src/services/web3auth.tsx) demo example.

**Comment by robport** [comment]
I've been through the next-app version, which is structured the same way,
and it doesn't distinguish between not_ready and ready.

This means you get a connect button when it's not_ready (when it should say
'Initialising...' or display spinner or something

**Reply by irfanonk** [reply]
I have created 3 state: isInitialized:false, isLoading:true and user:null

```
      try {
     
      // config here

       subscribeAuthEvents(web3AuthInstance);
        web3AuthInstance.configureAdapter(adapter);
        await web3AuthInstance.init();

      } catch (error) {
        console.error(error);
      } finally {
        setIsInitialized(true);
        // ensure that loading status toggle to false
        // if there is not a user logged in
        setTimeout(() => {
          if (isLoading) setIsLoading(false);
        }, 1500);
      }

 const subscribeAuthEvents = (web3auth) => {
      // Can subscribe to all ADAPTER_EVENTS and LOGIN_MODAL_EVENTS
      web3auth.on(ADAPTER_EVENTS.CONNECTED, async (data) => {
        console.log('successfully logged in', data);
    const user= await web3auth.getUserInfo();
    setUser(user)
}

```
